feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/waiting/WaitingNovaCardTopUpFragment.kt

```diff
@@ -14,10 +15,15 @@ class WaitingNovaCardTopUpFragment : BaseBottomSheetFragment<WaitingNovaCardTopU
 
     override fun initViews() {
         dialog?.setCanceledOnTouchOutside(false)
-        isCancelable = false
         getBehaviour().isHideable = false
     }
 
+    override fun onCancel(dialog: DialogInterface) {
+        viewModel.closeNovaCard()
```

Comment:
Fragment should not command ViewModel but rather notify it about events. I think we should rename viewmodel's method to smth like "onScreenCancelled`